*WIP*: Restructure Routes and Controllers

Mister-Corn opened the pull request:
# Main Goals
* [x] Restructure Routes and Controllers

Reafactoring the handler callback functions to be declared separately in another file and passed appropriately to express.

This is to improve readability and make unit testing easier.

* [x] Implement some tests for model and controller

Implement unit tests to test CRUD/query of database and controller ability to formulate proper responses.

Mister-Corn committed:
Start restructuring Routes and Controllers

Reafactoring the handler callback functions to be declared separately in another file and passed appropriately to express.

This is to improve readability and make unit testing easier.

Additional minor changes:
* Made styling more consistent in various parts.
* Added jest and supertest in prep for testing.

Mister-Corn committed:
Create initial tests for product handler functions

Created tests as a template for future tests for handlers. A lot of the mocking boilerplate is in place, so it should be easier to write up tests in the future.

Also:
* Tweaked structure of "controller" and "routes" folders. Individual parts are in their own file, and index.js gathers them together and module.exports them.

Mister-Corn committed:
Make minor changes to test formatting

Mister-Corn committed:
Implement test for model and DB

Mocked the DB for the test DB in this test. We can use this pattern for other model function tests.

NandoTheessen commented on controller/tests/products.spec.js:
```diff
@@ -1,13 +1,62 @@
+    // Mock db call in file
+    db.mockImplementation((table) => {
+      return test_db(table);
+    });
```
Uncaught promises are deprecated.
If you want to use async / await, you'll have to use the 
```
try {
} catch (err) {
}
```
Syntax

Mister-Corn replied:
You've also suggested something like this:

```
const x = await someFunction().catch(err=>whatever);
```

I find that a lot more elegant instead of doing try/catch for just one await statement, so I'm going to roll with that. Thanks for the feedback!

Mister-Corn committed:
Handle rejected promises approriately.

Some await statements did not have a way to handle rejections, so these statements now either have a catch  or are within a try/catch block.
